File: Medium/Java/120. Triangle.java
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        int[] path = new int[triangle.size() + 1];

        for(int i = triangle.size() - 1; 0 <= i; i--){
            for(int j = 0; j < triangle.get(i).size(); j++){
                path[j] = Math.min(path[j], path[j + 1]) + triangle.get(i).get(j);
            }
        }

        return path[0];
    }
}

File: Medium/JavaScript/120. Triangle.js
var minimumTotal = function(triangle) {
    for(let i = triangle.length - 2; 0 <= i; i--){
        for(let j = triangle[i].length - 1; 0 <= j; j--){
            triangle[i][j] += Math.min(triangle[i + 1][j], triangle[i + 1][j + 1]);
        }
    }

    return triangle[0][0];
};
